### PR TITLE
Features/confirm account

### PR DESCRIPTION
**Pull request by santir489**
This PR includes the "mail confirmation account" feature

### CODE REVIEW

File: db/migrate/20190612144842_add_user_attributes.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class AddUserAttributes < ActiveRecord::Migration[5.2]
   def change
-    add_column          :users, :gender, :integer 
-    change_column_null  :users, :gender, false   
+    add_column          :users, :gender, :integer
```
Comment (JuanfraM) [inline db/migrate/20190612144842_add_user_attributes.rb:3]: You can use null: false and avoid the next line:
add_column          :users, :gender, :integer, null: false

Same for the next cases

**Review by JuanfraM — APPROVED** [review]
Good PR!
Just one comment

File: config/initializers/devise_token_auth.rb [hunk 2]
```diff
@@ -1,50 +1,6 @@
...
-  #                        :'client' => 'client',
-  #                        :'expiry' => 'expiry',
-  #                        :'uid' => 'uid',
-  #                        :'token-type' => 'token-type' }
-
-  # By default, only Bearer Token authentication is implemented out of the box.
-  # If, however, you wish to integrate with legacy Devise authentication, you can
-  # do so by enabling this flag. NOTE: This feature is highly experimental!
-  # config.enable_standard_devise_support = false
+  config.default_confirm_success_url = 'http://localhost:3000/api/v1/user'
```
Comment (mdornel) [inline config/initializers/devise_token_auth.rb]: move `http://localhost:3000/api/v1/user` to a `ENV` variable, the host part at least.

**Review by mdornel — COMMENTED** [review]
As good practice, always that we introduce a new feature we include test for it as well. So, it would be great if you can add some testing for confirm account feature.

File: db/migrate/20190612144842_add_user_attributes.rb (hunk 1 above)
Comment (santir489) [reply db/migrate/20190612144842_add_user_attributes.rb:3]: Yes, thats makes sense. But sqlite complains when trying to add a column with not null constraint and no default value.

http://www.rhuberdeau.com/articles/18-Rails-SQLite3-SQLException-Cannot-add-a-NOT-NULL-column-with-default-value-NULL

Comment (JuanfraM) [reply db/migrate/20190612144842_add_user_attributes.rb:3]: Ok got it 👍

File: config/environments/development.rb [hunk 3]
```diff
@@ -51,4 +51,8 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+
+  config.action_mailer.default_url_options = { :host => 'localhost:3000' }
```
Comment (GuilleLeopold) [inline config/environments/development.rb:57]: Use newer syntax

File: db/migrate/20190612144842_add_user_attributes.rb (hunk 1 above)
Comment (GuilleLeopold) [reply db/migrate/20190612144842_add_user_attributes.rb:3]: We are migrating to Postgresql in the next PR

**Review by GuilleLeopold — APPROVED** [review]
One small comment. Let me know when you upload the confirmation tests

**Review by mdornel — APPROVED** [review]

**Review by mdornel — APPROVED** [review]

File: config/environments/production.rb [hunk 4]
```diff
@@ -74,6 +74,18 @@
   # require 'syslog/logger'
   # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
 
+  config.action_mailer.delivery_method = :smtp
+
+  ActionMailer::Base.smtp_settings = {
+    :user_name => ENV['SENDGRID_USERNAME'],
```
Comment (jdrosales17) [inline config/environments/production.rb]: Use the new sintax here as well

File: spec/request/api/v1/confirmations/show_confirmation_spec.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+describe 'GET /api/v1/auth/confirmation', type: :request do
+    let(:user) { create(:user, { password: 'password' } ) }
+
+    it 'returns the user is not confirmed' do
```
Comment (jdrosales17) [inline spec/request/api/v1/confirmations/show_confirmation_spec.rb]: This is not testing any logic, I would remove this test

File: spec/request/api/v1/confirmations/show_confirmation_spec.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      expect(user.confirmed?).to be false
+    end
+
+    it 'returns that a confirmation email was sent' do
+      post user_session_path, params: {email: user.email, password: "password"}, as: :json
```
Comment (jdrosales17) [inline spec/request/api/v1/confirmations/show_confirmation_spec.rb]: I would move this to another request test as it is part of the sign in

File: spec/request/api/v1/confirmations/show_confirmation_spec.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+describe 'GET /api/v1/auth/confirmation', type: :request do
```
Comment (jdrosales17) [inline spec/request/api/v1/confirmations/show_confirmation_spec.rb:3]: I would also test that the confirmation email is sent, but on another file as it is part of the registration process, with something like:
```
expect { post registration_path, params: params, as: :json }
  .to change { ActionMailer::Base.deliveries.count }.by(1)
```

File: config/environments/production.rb (hunk 4 above)
Comment (JuanfraM) [reply config/environments/production.rb]:
```
ActionMailer::Base.smtp_settings = {
 address: 'smtp.sendgrid.net',
 password: ENV['SENDGRID_PASSWORD']
}
```

File: spec/request/api/v1/confirmations/show_confirmation_spec.rb (hunk 6 above)
Comment (JuanfraM) [reply spec/request/api/v1/confirmations/show_confirmation_spec.rb]: In the describe of the test you said you are testing a `GET` not a `POST`

Comment (JuanfraM) [reply spec/request/api/v1/confirmations/show_confirmation_spec.rb]: Also, use single quotes for simple strings

**Pull request merged**